scvi/inference/posterior.py

```diff
@@ -1157,7 +1179,7 @@ def get_sample_scale(self, transform_batch=None):
                             y=labels,
                             n_samples=1,
```

Comment:
Should we allow many samples here and take the mean? In general it would be nice if these `get` methods could return samples and expected value estimates.

Reply:
If so, we could increase n_samples here (could lead to memory issues in some cases), or put a for loop around everything and take the average.

Reply:
It's a bit messy but I added a number of options for getting samples.